FirebaseAppCheck/Sources/Core/FIRAppCheck.m

```diff
@@ -330,6 +375,24 @@ - (nonnull NSString *)notificationTokenKey {
       });
 }
 
+- (FBLPromise<FIRAppCheckToken *> *)limitedUseToken {
+  return
+      [FBLPromise wrapObjectOrErrorCompletion:^(
+                      FBLPromiseObjectOrErrorCompletion _Nonnull handler) {
+        [self.appCheckProvider getTokenWithCompletion:handler];
+      }].then(^id _Nullable(FIRAppCheckToken *_Nullable token) {
+        // TODO: Make sure the self.tokenRefresher is updated only once. Currently the timer will be
+        // updated twice in the case when the refresh triggered by self.tokenRefresher, but it
+        // should be fine for now as it is a relatively cheap operation.
+        __auto_type refreshResult = [[FIRAppCheckTokenRefreshResult alloc]
+            initWithStatusSuccessAndExpirationDate:token.expirationDate
+                                    receivedAtDate:token.receivedAtDate];
+        [self.tokenRefresher updateWithRefreshResult:refreshResult];
+        [self postTokenUpdateNotificationWithToken:token];
+        return token;
+      });
```

Comment:
This `then` clause looks like it handles token refresh and updating listeners -- for `limitedUseTokenWithCompletion`, we'll just be returning an App Check token without side-effects (i.e. refresh, listeners, storage), so you can remove this section.

Reply:
thx, removed the refresh and updating.